avaframe/com1DFAPy/DFAfunctionsCython.pyx

```diff
@@ -344,6 +341,19 @@ def computeForceC(cfg, particles, fields, dem, dT):
   particles['uz'] = np.asarray(UZ)
   particles['m'] = np.asarray(mass)
 
+  # update mass available for entrainement
+  # TODO: this allows to entrain more mass then available...
+  for j in range(Npart):
+    indCellX = IndCellX[j]
+    indCellY = IndCellY[j]
+    entrMassCell = entrMassRaster[indCellY, indCellX]
+    areaCell = areaRatser[indCellY, indCellX]
+    dm = dM[j]
+    # update surfacic entrainment mass available
+    entrMassCell = entrMassCell - dm/areaCell
+    if entrMassCell < 0:
```

Comment:
How is it possible for entrMass to go below 0? Just little inaccuracies?

Reply:
nop i guess it can be bigger... in samos the mass available for entrainment is checked only at the end of the time step... which means you can entrain more snow than their is